src/time/measure.cr

```diff
@@ -0,0 +1,72 @@
+require "crystal/system/time"
+
+struct Time
+  # Measure elapsed time.
+  #
+  # Time measurement relies on a monotonic clock, that should be independent to
+  # time fluctuations, such as leap seconds or manually changing the computer
+  # time.
+  struct Measure
+    private getter seconds : Int64
+    private getter nanoseconds : Int32
+
+    # Starts a clock to measure elapsed time, or repeatedly report elapsed time
+    # since an initial start time.
+    def initialize
+      @seconds, @nanoseconds = Crystal::System::Time.monotonic
+    end
+
+    # Returns the time span since the clock was started.
+    #
+    # ```
+    # timer = Time::Measure.new
+    #
+    # loop do
+    #   do_something
+    #   p timer.elapsed # => 00:00:01.000000023
+    # end
+    # ```
+    def elapsed
+      seconds, nanoseconds = Crystal::System::Time.monotonic
+      Time::Span.new(seconds: seconds - @seconds, nanoseconds: nanoseconds - @nanoseconds)
+    end
+
+    # Returns true once *span* time has passed since the clock was created.
```

Comment:
Maybe it's better to use `if` instead of `once`. It feels to me like it might suggest this could be blocking *until* that amount of time has passed.